src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java

```diff
@@ -3403,15 +3404,17 @@ private Principal resolvePrincipal(Integer id) throws IOException {
           log.log(Level.FINE, "Member id {0} is not available in memberid"
               + " mapping for Web [{1}] under Site Collection [{2}].",
               new Object[] {id, webUrl, siteUrl});
-          if (siteUserMapping == null) {
+          if (siteUserMapping == null && !siteUserMappingInitialized) {
             try {
               siteUserMapping = getSiteUserMapping();
               principal = siteUserMapping.getPrincipal(id);
             } catch (IOException ex) {
-              log.log(Level.FINE, "Could not resolve site user mapping for "
+              log.log(Level.WARNING, "Could not resolve site user mapping for "
                   + id, ex);
+            } finally {
+              siteUserMappingInitialized = true;
             }
-          } else {
+          } else if (!siteUserMappingInitialized) {
```

Comment:
This seems like the wrong condition. We need siteUserMapping != null here. In that case, I would change the else if to just an if, and drop the getPrincipal call inside the then block (basically take the other path from the one we chose before).